### PR TITLE
Polish api Program/CompiledProgram/ParallelEnv doc & code example

### PR DESCRIPTION
**Pull request by chenwhql**
### PR types

Others
### PR changes

Docs
### Describe

Polish api Program/CompiledProgram/ParallelEnv doc & code example (only change doc)

cn doc: https://github.com/PaddlePaddle/FluidDoc/pull/2715

image

image
image

### CODE REVIEW

File: python/paddle/fluid/framework.py [hunk 1]
```diff
@@ -4338,33 +4345,39 @@ def clone(self, for_test=False):
...
+            .. note::
+                The Program's order maybe different after :code:`clone` and
+                this will not affect your training or testing progress. In the following
+                example we give you an simple method :code:`print_prog(program)` to
+                print Program Descs inorder to make sure you have same print result
+                after :code:`clone`:
 
-                import paddle.fluid as fluid
-                import six
+                .. code-block:: python
```
Comment (TCChenlong) [inline python/paddle/fluid/framework.py]: 这里缩进有问题哈 预览里和note的块合在一起了

File: python/paddle/fluid/framework.py [hunk 2]
```diff
@@ -4294,17 +4303,15 @@ def _version(self):
...
+        .. note:::
+            1.** :code:`Program.clone()` **method DOES NOT clone** :ref:`api_paddle_io_DataLoader` . 
+            2. Recommend you to use** :code:`clone` **before using** :code:`Opimizer.minimize` . 
+            3. This API has no effect in Dygraph Mode** 
 
         Create a new Program with forward content of original one when ``for_test=True``.
         Create a new Program as same as the original one when ``for_test=False``.
 
-        Some operators, e.g., :ref:`api_fluid_layers_batch_norm` , behave differently between
+        Some operators, e.g., :ref:`api_paddle_fluid_layers_batch_norm` , behave differently between
```
Comment (TCChenlong) [inline python/paddle/fluid/framework.py:4314]: 4325 - 4335 没有更新哈

File: python/paddle/fluid/framework.py [hunk 3]
```diff
@@ -5012,44 +5055,25 @@ def all_parameters(self):
         Examples:
             .. code-block:: python
 
-                import paddle.fluid as fluid
+                import paddle
```
Comment (TCChenlong) [inline python/paddle/fluid/framework.py:5062]: 这个截图里 没有更新 但是代码里更新了 是先截的图么？

File: python/paddle/fluid/compiler.py [hunk 4]
```diff
@@ -93,42 +93,45 @@ class CompiledProgram(object):
...
+            place = paddle.CUDAPlace(0) # paddle.CPUPlace()
+            exe = static.Executor(place)
 
-          exe.run(fluid.default_startup_program())
-          compiled_prog = fluid.CompiledProgram(
-                   fluid.default_main_program())
+            data = static.data(name='X', shape=[None, 1], dtype='float32')
+            hidden = static.nn.fc(input=data, size=10)
+            loss = paddle.mean(hidden)
+            paddle.optimizer.SGD(learning_rate=0.01).minimize(loss)
```
Comment (TCChenlong) [inline python/paddle/fluid/compiler.py:125]: 这里明确下 是minimize(loss) 还是 step()?

File: python/paddle/fluid/compiler.py [hunk 5]
```diff
@@ -214,50 +220,53 @@ def with_data_parallel(self,
...
+                    os.environ['CPU_NUM'] = str(2)
+
+                exe = static.Executor(place)
+
+                data = static.data(name='X', shape=[None, 1], dtype='float32')
+                hidden = static.nn.fc(input=data, size=10)
+                loss = paddle.mean(hidden)
+
+                test_program = static.default_main_program().clone(for_test=True)
+                paddle.optimizer.SGD(learning_rate=0.01).minimize(loss)
```
Comment (TCChenlong) [inline python/paddle/fluid/compiler.py:250]: 同上

File: python/paddle/fluid/compiler.py (hunk 4 above)
Comment (chenwhql) [reply python/paddle/fluid/compiler.py:125]: 这些都是静态图API，静态图还是minimize，step是仅用于动态图的

File: python/paddle/fluid/compiler.py (hunk 5 above)
Comment (chenwhql) [reply python/paddle/fluid/compiler.py:250]: 同上

File: python/paddle/fluid/framework.py (hunk 3 above)
Comment (chenwhql) [reply python/paddle/fluid/framework.py:5062]: 是的，我再截一次

File: python/paddle/fluid/framework.py (hunk 2 above)
Comment (chenwhql) [reply python/paddle/fluid/framework.py:4314]: done, thx

File: python/paddle/fluid/framework.py (hunk 1 above)
Comment (chenwhql) [reply python/paddle/fluid/framework.py]: done, thx

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM
